kaiax/valset/impl/council.go

```diff
@@ -0,0 +1,186 @@
+package impl
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"sort"
+
+	"github.com/kaiachain/kaia/blockchain/types"
+	"github.com/kaiachain/kaia/common"
+	"github.com/kaiachain/kaia/kaiax/gov"
+	"github.com/kaiachain/kaia/kaiax/valset"
+	"github.com/kaiachain/kaia/params"
+)
+
+// council is used to calculate committee or proposer.
+// councilList(N) -> apply demoted rule -> demoted,qualified -> block proposed and apply votes -> councilList(N+1)
+type council struct {
```

Comment:
`council` struct contains the after concepts of qualifying process. (newCouncil equals qualify). Why the function is named as newCouncil is because it's a constructor of Council structure. So, I'll explicitly change the name.

About proposer, committee, instead of using config, CommitteeContext is implementing now and it can replace the config. CommitteeContext will includes all necessary information and have methods(selectProposer, selectCommittee).